store/PoolInstancesSlice/index.tsx

```diff
@@ -140,50 +146,103 @@ export const createPoolsInstancesSlice: StateSlice<IPoolsInstancesSlice> = (set,
 
     handlePoolUpkeep: (pool, provider, account) => {
         get().setPoolIsWaiting(pool, false);
-        get().updateTokenBalances(pool, provider, account);
+        get().updatePoolTokenBalances([pool], provider, account);
+        get().updateSettlementTokenBalances([pool], provider, account);
         get().updatePoolBalances(pool, provider);
     },
 
-    updateTokenBalances: (pool_, provider, account) => {
-        if (!provider || !account || !get().pools[pool_]) {
+    updateSettlementTokenBalances: (pools_, provider, account) => {
+        if (!provider || !account) {
             return false;
         }
-        const pool = get().pools[pool_].poolInstance;
-        const tokens = [pool.shortToken.address, pool.longToken.address, pool.settlementToken.address];
-        const decimals = pool.settlementToken.decimals;
-        fetchTokenBalances(tokens, provider, account, pool.address)
+        const uniqueSettlementTokensMap: Record<string, string[]> = Object.values(get().pools).reduce(
+            (
+                o,
+                {
+                    poolInstance: {
+                        address,
+                        settlementToken: { address: tokenAddress },
+                    },
+                },
+            ) => {
+                o[tokenAddress] = o[tokenAddress] || [];
+                o[tokenAddress].push(address);
+                return o;
+            },
+            {} as Record<string, string[]>,
+        );
+        const relevantSettlementTokens = Object.keys(
+            pools_.reduce((o, pool) => {
+                const settlementToken = get().pools[pool]?.poolInstance?.settlementToken?.address;
+                if (settlementToken) {
+                    o[settlementToken] = true;
+                }
+                return o;
+            }, {} as Record<string, boolean>),
+        );
```

Comment:
this could potentially be simplified to something like:

```
relevantSettlementTokenMap = {}
for (const pool in get()) {
    relevantSettlementTokenMap[pool.settlementToken.address] = true
}
relevantSettlementTokenAddresses = Object.keys(relevantSettlementTokenMap)
```

Reply:
just needs to factor in the pools_ param

Reply:
how about

```
relevantSettlementTokenMap = {}
for (const pool in get()) {
    if(pools_.includes(pool.address)) {
       relevantSettlementTokenMap[pool.settlementToken.address] = true
    }
}
relevantSettlementTokenAddresses = Object.keys(relevantSettlementTokenMap)
```

Reply:
yep but then it doesnt get 'all' relevant, just gets relevant based off pools_. Needs to also find all pools amongst get() which is why the two loops. Will try reduce so its a bit cleaner.

Reply:
doesn't it just need the unique `settlementToken` between all pools specified in `pools_`?